osler/core/models.py

```diff
@@ -536,3 +566,48 @@ def admin_url(self):
     def __str__(self):
         return " ".join(["AI for", str(self.patient) + ":",
                          str(self.instruction), "due on", str(self.due_date)])
+
+
+class EncounterStatus(models.Model):
+    '''Different status for encounter, as simple as Active/Inactive 
+    or Waiting/Team in Room/Attending etc'''
+    name = models.CharField(max_length=100, primary_key=True)
+    is_active = models.BooleanField(default=False)
+
+    def __str__(self):
+        return self.name
+
+
+def default_active_status():
+    status, created = EncounterStatus.objects.get_or_create(
+        name=settings.OSLER_DEFAULT_ACTIVE_STATUS[0],
+        is_active=settings.OSLER_DEFAULT_ACTIVE_STATUS[1])
+    return status
+
+
+def default_inactive_status():
+    status, created = EncounterStatus.objects.get_or_create(
```

Comment:
Urh so the data migration only creates an instance the default inactive one to populate the existing workups. But presumably one day we could squash(?) the migrations and not have EncounterStatuses made yet when something is started up? It's like how we have to make Genders?

I do get_or_create for when this is the first time you are making the default_inactive_status. So if one day you ever change what you set the default inactive one as, it will just make a new EncounterStatus and assign that going forward, leaving the old ones as is?

Reply:
Interesting. Okay. That makes sense to me!